projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.ts

```diff
@@ -48,14 +55,36 @@ export class DocumentEditorComponent {
    * @param pid string - identifier of the external record
    */
   importFromExternalSource(source: string, pid: string) {
-    this.editorService.getRecordFromExternal(source, pid).subscribe(
+    this._editorService.getRecordFromExternal(source, pid).subscribe(
       record => {
         if (record) {
           this.model = record.metadata;
         } else {
-          this.toastrService.warning(
-            this.translateService.instant('Does not exists on the remote server!'),
-            this.translateService.instant('Import')
+          this._toastrService.warning(
+            this._translateService.instant('Does not exists on the remote server!'),
+            this._translateService.instant('Import')
+          );
+        }
+      }
+    );
+  }
+
+  /**
+   * Get record by type and pid
+   * @param type string - resource type
+   * @param pid - resource pid
+   */
+  duplicateRecord(type: string, pid: string) {
+    this._editorService.getRecord(type, pid).subscribe(
```

Comment:
why not using the recordService directly?

Reply:
done